src/containers/PDiskPage/DecommissionButton/DecommissionButton.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+
+import {ChevronDown} from '@gravity-ui/icons';
+import type {DropdownMenuItem, DropdownMenuProps} from '@gravity-ui/uikit';
+import {Button, DropdownMenu, Icon, Popover} from '@gravity-ui/uikit';
+
+import {CriticalActionDialog} from '../../../components/CriticalActionDialog/CriticalActionDialog';
+import {isErrorWithRetry} from '../../../components/CriticalActionDialog/utils';
+import type {EDecommitStatus} from '../../../types/api/pdisk';
+import {wait} from '../../../utils';
+import {cn} from '../../../utils/cn';
+import {pDiskPageKeyset} from '../i18n';
+
+import decommissionIcon from '../../../assets/icons/decommission.svg';
+
+import './DecommissionButton.scss';
+
+const b = cn('ydb-pdisk-decommission-button');
+
+function getDecommissionWarningText(decommission?: EDecommitStatus) {
+    if (decommission === 'DECOMMIT_IMMINENT') {
+        return pDiskPageKeyset('decommission-dialog-imminent-warning');
+    }
+    if (decommission === 'DECOMMIT_PENDING') {
+        return pDiskPageKeyset('decommission-dialog-pending-warning');
+    }
+    if (decommission === 'DECOMMIT_REJECTED') {
+        return pDiskPageKeyset('decommission-dialog-rejected-warning');
+    }
+    if (decommission === 'DECOMMIT_NONE') {
+        return pDiskPageKeyset('decommission-dialog-none-warning');
+    }
+
+    return undefined;
+}
+
+interface DecommissionButtonProps {
+    decommission?: EDecommitStatus;
+    onConfirmAction: (newDecomissionStatus?: EDecommitStatus, isRetry?: boolean) => Promise<void>;
+    onConfirmActionSuccess: (() => Promise<void>) | VoidFunction;
+    buttonDisabled?: boolean;
+    popoverDisabled?: boolean;
+}
+
+export function DecommissionButton({
+    decommission,
+    onConfirmAction,
+    onConfirmActionSuccess,
+    buttonDisabled,
+    popoverDisabled,
+}: DecommissionButtonProps) {
+    const [newDecommission, setNewDecommission] = React.useState<EDecommitStatus | undefined>();
+    const [buttonLoading, setButtonLoading] = React.useState(false);
+    const [withRetry, setWithRetry] = React.useState(false);
+
+    const handleConfirmAction = async (isRetry?: boolean) => {
+        setButtonLoading(true);
+        await onConfirmAction(newDecommission, isRetry);
+        setButtonLoading(false);
+    };
+
+    const handleConfirmActionSuccess = async () => {
+        setWithRetry(false);
+
+        if (onConfirmActionSuccess) {
+            setButtonLoading(true);
+
+            // Decommission needs some time to change
+            // Wait for some time to send request for updated data
+            await wait(5000);
```

Comment:
This assumes the decommission status will change within 5 seconds, which may not always be true. A better approach would be to poll the API until the status changes or a timeout is reached.

Reply:
I don't like the idea to poll the handler, that collects entity data (`/pdisk/info` with data both from BSC and Node Whiteboard is the only way we can get decommit status now). 

So this approach is not the best, but works in most of the cases - in other cases you could get actual status after some more time with refresh button